pkg/apis/certificates/helpers.go

```diff
@@ -118,7 +130,7 @@ func ValidateKubeletClientCSR(req *x509.CertificateRequest, usages sets.String)
 		return commonNameNotSystemNode
```

Comment:
ValidateKubeletClientCSR and ValidateKubeletServingCSR are called by both the controller and APIServer defaulting... in 1.25 we only want to modify behavior of the controller. One way to do that would be to plumb an `allowOmittingUsageKeyEncipherment bool` option through all callers and only pass true from the signer/approver controller call sites in 1.25

Reply:
- `IsKubeletClientCSR`(in pkg/apis/certificates/helpers.go) will keep as before
- `isKubeletClient`(pkg/controller/certificates/signer/signer.go) will tolerate missing `UsageKeyEncipherment`

Reply:
client changes look good once https://github.com/kubernetes/kubernetes/pull/111061/files#r935588239 is addressed

the IsKubeletServingCSR / ValidateKubeletServingCSR functions need the same treatment

Reply:
Fixed. They can be more consistent.